lib/sssd.c

```diff
@@ -43,15 +47,10 @@ sssd_flush_cache(int dbflags)
 
 	p = stpcpy(sss_cache_args, "-");
 	if (dbflags & SSSD_DB_PASSWD)
-		stpcpy(p, "U");
+		p = stpcpy(p, "U");
 	if (dbflags & SSSD_DB_GROUP)
-		stpcpy(p, "G");
+		p = stpcpy(p, "G");
 
-	if (streq(p, "")) {
-		/* Neither passwd nor group, nothing to do */
-		free(sss_cache_args);
-		return 0;
-	}
```

Comment:
Oops, I did it very wrongly.  Thanks for the fix!

I should probably issue a stable release 4.18.1 with this fix.

Would you mind reformatting the commit message to refer to the fixed commit this way?:

```
Fixes: 59e5eef38f89 (2024-07-03; "contrib, lib/, src/, tests/: Use stpcpy(3) instead of its pattern")
```

Also:

```
Reviewed-by: Alejandro Colomar <alx@kernel.org>
```
(this is only for the first commit.)

Reply:
I guess I'll not do a stable release just for this then.  If anyone is interested in it, they can ask for it, though.